cmd/subcmds/sg.go

```diff
@@ -24,8 +25,8 @@ func NewSGCommand(args *inArgs) *cobra.Command {
 			if err != nil {
 				return err
 			}
-			collection := synth.MakeSG(spec)
-			return writeOutput(args, collection, &spec.Defs.ConfigDefs)
+			sgSynthesizer := synth.SGSynthesizer{Spec: spec, Result: ir.NewSGCollection()}
```

Comment:
Might be nicer to provide a `NewSGSynthesizer(spec *ir.Spec)` method.
This allows to un-export the internal `Spec` and `Result` members, and doesn't require the caller to know how to initialize them.

Reply:
How about having a common interface for both synthesizers?
```golang
type Synthesizer interface {
	Synth() ir.Collection
}
```
Then you can have a single function to implement common code in the `RunE` of both `sg` and `acl` sub-commands.

Reply:
done